### ISSUE TITLE
Locked and Unlocked states should have a different icon

### ISSUE THREAD

**Issue by NoXPhasma** [PaulCombal/SamRewritten-legacy#44]
Currently both states for locked and unlocked achievements have the almost same icon, this makes it not very easy to see which one is already unlocked. I think the clarity would be much better if they wouldn't share such a similar icon.

![state](https://user-images.githubusercontent.com/7719590/70669224-0727f800-1c76-11ea-8255-7f9c6b41e770.png)

**Comment by PaulCombal** [comment]
You're right.. Can you think of an emoji to use for locked / unlocked / To unlock / To relock?

**Comment by NoXPhasma** [comment]
As emoji there is not much choice. Maybe with a key for locked 🔐 and unlocked 🔓

**Comment by NoXPhasma** [comment]
Or maybe a 🔒 for locked and a ✅ for unlocked. That at least would be more than clean.

**Comment by PaulCombal** [comment]
I think it would be better to have something looking consistent, I like 🔐 for locked better for example.
How do you feel about ✔ to unlock, and ✅ to relock?

EDIT: to unlock would appear green like here https://emojipedia.org/heavy-check-mark/

**Comment by NoXPhasma** [comment]
That might work.

**Comment by wgpierce** [comment]
I don't like the check marks; they convey even less info. We could also include custom images if that would help.

**Comment by PaulCombal** [comment]
Do you have any ideas for the images?

**Comment by wgpierce** [comment]
The original SAM uses the [Fugue Icons](https://p.yusukekamiyamane.com/) set, which has a nice lock and unlock.

**Comment by PaulCombal** [comment]
I'm not a big fan of that to be fair. Not that they don't look good or anything but the reason why I chose Gtk is to create an app that integrates nicely with the user's desktop.
If going with images seems to be the solution, a quick tour in my system icon pack resulted in this.
![image](https://user-images.githubusercontent.com/5382590/70787080-43e71280-1d8e-11ea-846a-a2989594c34f.png)
How about that?

**Comment by NoXPhasma** [comment]
Those are dependant on the Icon theme the user has selected on their system and that should be preferable.

**Comment by PaulCombal** [comment]
This is now corrected on the master branch, can you tell me if you still have issues with the new release?

**Comment by NoXPhasma** [comment]
I can not, as the build fails. Here's the log: https://gist.github.com/NoXPhasma/d5898e3cff5dcde83b7ae986169fc771

This is on Linux Mint 19.3, GCC 9

**Comment by PaulCombal** [comment]
I updated the Readme, I ported the library from Gtk to Gtkmm, so you might need to install this library beforehand

**Comment by NoXPhasma** [comment]
I've done that, all packages listed in the readme are installed, including `libgtkmm-3.0-dev`. It breaks for a different reason, which I don't know.

Gtk version here is 3.22.

**Comment by NoXPhasma** [comment]
Apparently the issue is that the version of Gtkmm in the Ubuntu 18.04 repositories is missing the function `make_managed`. This won't build.

**Comment by PaulCombal** [comment]
The best way would be to release an Appimage build so you could try out, but Ubuntu's glibc is too old for them to run if I build them myself. I'll create a VM to build the Appimage when I'll have time, I'll keep you updated. Do you happen to know if this would work on the newest Ubuntu versions?

**Comment by NoXPhasma** [comment]
Latest might work, as it has Gtk 3.24. But it also has a newer Glibc version.

**Comment by PaulCombal** [comment]
Ok, Ijust built an Appimage on 19.04, and it worked almost out of the box. I've uploaded it to the 201912 release. Can you run it? If not I think that sadly 18.04 will not be supported at all..

**Comment by NoXPhasma** [comment]
Doesn't work due to missing GLIBC.

**Comment by PaulCombal** [comment]
What's the error exactly? Isn't it something about a version mismatch?

**Comment by NoXPhasma** [comment]
```
─$ ./SamRewritten-ab06374-x86_64.AppImage 
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libgtk-3.so.0)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libldap_r-2.4.so.2)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/liblber-2.4.so.2)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libcairo.so.2)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libpng16.so.16)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libsystemd.so.0)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libsqlite3.so.0)
/tmp/.mount_SamRewdD27Cr/usr/bin/samrewritten: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/.mount_SamRewdD27Cr/usr/bin/../lib/libsqlite3.so.0)
```

**Comment by PaulCombal** [comment]
Exactly.. Well I think that signs the end of support for 18.04, hard to believe it's nearly a 2 years old stable release, so that is expected.. For now the only solution I can think of is using older builds unfortunately.
I can't say efforts will be made to make it more backwards compatible since the second GTK4 will be out I will start using it for technical / performance reasons, further getting away from the older LTS stack

**Comment by NoXPhasma** [comment]
There would still flatpak, by that it would run on all distributions. No idea if that is a possibility for you.

Anyway, I will not upgrade my system until Linux Mint 20 will be available, so I'll stay on an older build for now.

**Comment by PaulCombal** [comment]
I'm open to releasing a Flatpak version, if I'm getting a pull request for that.
I'll leave this issue open for a bit, please give us an update when you have time to test the new versions!

**Comment by wgpierce** [comment]
I've created an appimage for Ubuntu 18.04 in releases with the hacks needed for that platform. Can you try that out? From my end, looks like this issue is resolved at latest master.

**Comment by NoXPhasma** [comment]
This works fine, thank you!

**Comment by wgpierce** [comment]
Cool, issue is fixed by commit 934ce004, closing.
